README.md
# Iranian Bank Logos
[![Gitter][gitter-badge]][gitter]
[![Codacy][codacy-badge]][codacy]
[![Releases][releases-badge]][releases]
[![Download][download-badge]][download]

&#x202b;این کتابخانه شامل فهرست کاملی از لوگو یا نمایه های بانک های ایران است که (تا لحظه انتشار این نسخه) بر روی سایت بانک مرکزی ایران منتشر شده است. 
&#x202b;لطفا در صورت مشاهده مغایرت در رنگ، شکل، فرم یا محتوای لوگوها همین جا یا از سایت طراحی وب ایران ما را مطلع فرمایید.

🌟 If you ❤️ library, please star it! 🌟

# نحوه استفاده

### 1. دریافت فایل

&#x202b;ساده ترین روش، استفاده از [npm](https://www.npmjs.com/package/iranianbanklogos) است

```sh
npm install iranianbanklogos
```

```sh
yarn add iranianbanklogos
```

&#x202b;راه دیگر، استفاده از [Nuget](https://www.nuget.org/packages/IranianBankLogos) است

```sh
Install-Package IranianBankLogos
```

&#x202b;البته می توانید از [Bower](http://bower.io/) هم استفاده کنید

```sh
bower install iranianbanklogos
```

&#x202b;در غیر این صورت شما می توانید از مسیر `dist/ibl.min.css` فایل استایل را دانلود و در پروژه خود استفاده کنید.

### 2. لینک به استایل

&#x202b;با افزودن لینک این استایل به پروژه یا HTML کار تمام است.

&#x202b;در پروژه های مبتنی بر `React` می توانید خط زیر را اضافه کنید:

```javascript
import "iranianbanklogos/dist/ibl.css";
```

&#x202b;یا در پروژه های دات نت و مانند آن می توان لینک را در قسمت `<head>` قرار داد:

```html
<link rel="stylesheet" href="bower_components/iranianbanklogos/dist/ibl.min.css">
```

### 3. نمونه کد

&#x202b;با مراجعه به [سایت پروژه](https://webdesigniran.github.io/IranianBankLogos/) می توانید بر روی هر لوگو کلیک کنید تا کد `HTML` مربوطه در کلیپ.بورد. کپی شود. 

&#x202b;سپس می توانید این کد را در `HTML` خود `Paste` استفاده کنید

```html
<i class="ibl64 ibl-bsi"></i> بانک صادرات
<i class="ibl64 ibl-mellat"></i> بانک ملت
<i class="ibl64 ibl-tejarat"></i> بانک تجارت
<i class="ibl64 ibl-bmi"></i> بانک ملی
<i class="ibl64 ibl-sepah"></i> بانک سپه
<i class="ibl64 ibl-bki"></i> بانک کشاورزی
<i class="ibl64 ibl-parsian"></i> بانک پارسیان
```

&#x202b;هر لوگو در چهار سایز 32، 64، 128 و 256 قابل استفاده است.

&#x202b;برای استفاده از هر سایز کلاس مشخص کننده سایز آن را مطابق نمونه زیر انتخاب کنید

&#x202b;همچنین می توانید با مراجعه به [سایت پروژه](https://webdesigniran.github.io/IranianBankLogos/) بر روی هر لوگو کلیک کنید تا کد `HTML` مربوطه در کلیپ.بورد. کپی شود. 

```html
بانک مسکن سایز 32 <i class="ibl32 ibl-maskan"></i>
بانک مسکن سایز 64 <i class="ibl64 ibl-maskan"></i>
بانک مسکن سایز 128 <i class="ibl128 ibl-maskan"></i>
بانک مسکن سایز 256 <i class="ibl256 ibl-maskan"></i>
```

# امکانات
- &#x202b;فرمت تصاویر PNG24 بدون شفافیت است
- &#x202b;از هر لوگو دو نسخه رنگی و خاکستری در پوشه تصاویر موجود است
- &#x202b;لوگوها در سایز پایه 300 پیکسل و مربعی طراحی شده اند
- &#x202b;خروجی با کیفیت در سایزهای 32 و 64 و 128 و 256 آماده شده است
- &#x202b;لوگوها با حفظ تناسب در محدوده های یکسان (در میان خطوط راهنما) متحد الشکل شده اند
- &#x202b;منبع فایل های گرافیکی (فتوشاپ فایل ها) منتشر شده است

<img alt="نمایی از خطوط راهنما و محدود سازی ابعاد لوگوها"
     src="https://cloud.githubusercontent.com/assets/6195199/15833653/01982676-2c3d-11e6-8bbf-00c1310d73de.png" 
     width="62%" />

# نحوه تولید و انتشار
- با کمک از فایل فتوشاپ-لوگو یک لوگوی جدید می سازیم
- سپس فایل های تصاویر رو به روزآوری می کنیم
- بعد فایل فتوشاپ-اسپریت رو به روزآوری می کنیم
- با فوتوشاپ چهار اسپریت جدید تولید می کنیم
- در فولدر سورس،‌ فایل های اسپریت را به روزآوری می کنیم
- در فولدر سورس،‌ فایل استایل را به روزآوری می کنیم
- نتایج جدید را دستی از سورس به فولدر مقصد کپی می کنیم
- حالا فولدر مقصد را پابلیش می کنیم
- نسخه برنامه را افزایش می دهیم
- برای پابلیش در مسیر اصلی دستورات زیر را استفاده می کنیم
     - اولین بار `npm adduser`
     - اولین بار `npm whoami`
     - هر بار `npm publish --access public`
- بعد از آن نسخه دمو را هم اپدیت و از داخل پوشه دمو، پابلیش می کنیم
     - دستور `yarn deploy`
- نتایج حجم فولدر مقصد را اندازه گیری کرده و README.md را آپدیت می کنیم
- فایل SECURITY.md رو به روزآوری کنیم

# پشتیبانی
لطفا اگر نسخه تصویری بهتری برای یک نمایه / لوگو یافته اید
یا اگر مشکلی مشاهده کردید
یا اگر پیشنهادی دارید

- &#x202b;یا Issue بزنید
- &#x202b;یا Fork & Pull کنید
- &#x202b;یا حداقل یک تلفن یا موبایل بزنید به [طراحی وب ایران](http://webdesigniran.com)

# License

The project is dedicated to public and is free for all uses, commercial or otherwise.
Supported by [Web Design Iran](http://webdesigniran.com)

این پروژه تحت حمایت [طراحی وب ایران](http://webdesigniran.com) بوده و برای استفاده تجاری یا غیر تجاری، رایگان است

<!-- Definitions -->

[gitter-badge]: https://badges.gitter.im/webdesigniran/IranianBankLogos.svg

[gitter]: https://gitter.im/webdesigniran/IranianBankLogos?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge

[codacy-badge]: https://app.codacy.com/project/badge/Grade/6b6a81d6448c41b3903b220d70df3759

[codacy]: https://www.codacy.com/gh/amastaneh/IranianBankLogos/dashboard?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=amastaneh/IranianBankLogos&amp;utm_campaign=Badge_Grade

[releases-badge]: https://img.shields.io/badge/release-v2.0-blue.svg?style=flat

[releases]: https://github.com/webdesigniran/IranianBankLogos

[download-badge]: https://img.shields.io/badge/download-260k-brightgreen.svg?style=flat

[download]: https://github.com/webdesigniran/IranianBankLogos


CHANGELOG.md
# CHANGELOG

## v3.0
  * &#x202b; افزودن لوگوی جدید بانک رفاه
  * &#x202b; تغییر فولدر منابع 
  * &#x202b; افزودن نسخه ری اکت به سایت دمو

## v2.0

  * &#x202b; کاهش حجم استایل فشرده شده یا خروجی نهایی از 7 کیلوبایت به 6 کیلوبایت
  * &#x202b; تبدیل موتور استایل از LESS به SASS
  * &#x202b; افزودن اندازه 32 به استایل
  * &#x202b; افزودن اندازه 256 به استایل
  * &#x202b; حذف اندازه 300 از استایل
  * &#x202b; بازسازی کامل لوگوی بانک ایران ونزوئلا
  * &#x202b; کاهش حجم بیش از 9 درصدی با فشرده سازی loosless تصاویر sprite
  * &#x202b; انتقال استاندارد تصاویر sprite از PNG24 به PNG8-256 و کاهش حجم بیش از 50 درصد
  * &#x202b; افزودن تصویر sprite با سایز 256
  * &#x202b; افزودن تصویر sprite با سایز 32


## v1.0

  * &#x202b; تهیه نسخه تجمعی در سایزهای 64 و 128 و 300 پیکسلی
  * &#x202b; تهیه نسخه تجمعی بر اساس آمار تعداد کارت های صادر شده
  * &#x202b; تهیه نسخه خاکستری
  * &#x202b; تهیه نسخه با کیفیت و فرمت PNG24 در سایز مربعی 300 پیکسل و پس زمینه یکپارچه سفید
  * &#x202b; یکسان سازی ارتفاع و عنوان بانک در یک قالب استاندارد و متحدالشکل با اندازه 300 پیکسلی
  * &#x202b; جمع آوری نسخه های باکیفیت لوگوهای بانک های ایرانی



demo/src/pages/pageHome.js
import React from 'react';
import GitHubForkRibbon from 'react-github-fork-ribbon'
import packageDemo from './../../package.json'
import "iranianbanklogos/dist/ibl.css";


const PageHome = () => {
    const [hint, setHint] = React.useState();
    const bankList = ["bsi", "mellat", "tejarat", "bmi", "sepah", "bki", "parsian", "maskan", "rb", "en", "ansar", "bpi", "sb", "sina", "post", "ghbi", "tt", "shahr", "ba", "sarmayeh", "day", "hi", "iz", "kar", "tourism", "bim", "edbi", "me", "ivbb", "resalat", "miran", "melal", "rb2"]
    const fa2enDigits = s => s.replace(/[0-9]/g, w => String.fromCharCode(w.charCodeAt(0) + 1728))

    return <React.Fragment>

        <header className="row text-center mt-5">
            <h1 className="fw-bolder">لوگوی بانک های ایران</h1>
            <h1 className="fw-bolder">Iranian Bank Logos</h1>
            <section className="my-4">
                <div>فهرست لوگوهای بانک های ایران</div>
                <div>مناسب سازی شده برای سامانه های مبتنی بر وب و موبایل</div>
                <div>با کلیک بر روی هر لوگو کد HTML آن در حافظه کپی می شود</div>
            </section>
        </header>

        <main className="row text-center">
            <div className="col-12">
                <button className="btn btn-dark mx-1" href="https://www.github.com/webdesigniran/IranianBankLogos">کدها در GitHub</button>
                <button className="btn btn-primary mx-1" href="https://www.nuget.org/packages/IranianBankLogos/">دانلود از NuGet</button>
            </div>

            {[32, 64, 128, 256].map(size =>
                <React.Fragment key={size} >
                    <div className="col-12 mt-5">
                        <h2 className="fw-bolder">سایز {fa2enDigits(size.toString())} پیکسلی</h2>
                        {bankList.map((bank, bankIndex) =>
                            <React.Fragment key={bankIndex}>
                                <span
                                    className={`ibl${size} ibl-${bank} cursor-pointer mx-1`}
                                    onClick={() => {
                                        setHint({ size, bank })
                                        navigator.clipboard.writeText(`<i class="ibl${size} ibl-${bank}"></i>`)
                                    }}
                                />
                                {(((bankIndex + 1) % 9 === 0) && size < 256) && <br />}
                            </React.Fragment>
                        )}
                    </div>

                    {(hint && hint.size === size) &&
                        <React.Fragment>
                            <div className="col-4" />
                            <div className="col-4 mt-2">
                                <div className="card bg-secondary p-3">
                                    <code className="text-white">
                                        &lt;i class="<span className="text-warning">ibl{hint.size} ibl-{hint.bank}</span>"&gt;&lt;/i&gt;
                                    </code>
                                </div>
                            </div>
                            <div className="col-4" />
                        </React.Fragment>
                    }

                </React.Fragment>
            )}
        </main>

        <footer className="row text-center mt-5 mb-2 text-muted fs-8">
            <hr />
            <div className="text-nowrap">
                تلاشی با ❤ در <a className="text-decoration-none text-secondary" href="http://webdesigniran.com">طراحی وب ایران</a>
            </div>
            <div className="text-nowrap">
                {packageDemo.name} {packageDemo.version}
            </div>
        </footer>

        <GitHubForkRibbon position="right"
            color="black"
            href="//github.com/amastaneh/IranianBankLogos"
            target="_blank">Fork me on GitHub</GitHubForkRibbon>
    </React.Fragment>
}

export default PageHome